### PR TITLE
Funannotate compare: add missing dep

### PR DESCRIPTION
**Pull request by abretaud**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [x] - This PR does something else (explain below)

Yet another funannotate PR! I've just realised there was a missing dep for the compare tool... Adding it + corresponding tests

### CODE REVIEW

File: tools/funannotate/funannotate_compare.xml [hunk 1]
```diff
@@ -5,6 +5,7 @@
     </macros>
     <requirements>
         <expand macro="requirements" />
+        <requirement type="package" version="3.3.20200621">phyml</requirement>
```
Comment (bgruening) [inline tools/funannotate/funannotate_compare.xml:8]: should that not be part of the conda package?

Comment (abretaud) [reply tools/funannotate/funannotate_compare.xml:8]: oh yeah, probably... will look into it

**Comment by abretaud** [comment]
So I've recycled this PR to fix an error on an output name (not sure if a version bump is necessary here?) + left the new tests

**Comment by bgruening** [comment]
Please note, that the package in bioconda needs 30min to appear on all mirrors.

**Review by bgruening — APPROVED** [review]

**Pull request merged**